cmd/auto-updater/main.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"storj.io/storj/internal/sync2"
+)
+
+var (
+	rootCmd = &cobra.Command{
+		Use:   "auto-updater",
+		Short: "auto-updater for storage node",
+	}
+	runCmd = &cobra.Command{
+		Use:   "run",
+		Short: "Run the auto updater for storage node",
+		RunE:  cmdRun,
+	}
+
+	interval   string
+	versionURL string
+	binaryPath string
+)
+
+func init() {
+	rootCmd.AddCommand(runCmd)
+
+	runCmd.Flags().StringVar(&interval, "interval", "05s", "interval for checking the new version")
+	runCmd.Flags().StringVar(&versionURL, "version-url", "https://version.storj.io/", "version server URL")
+	runCmd.Flags().StringVar(&binaryPath, "binary-path", "", "the storage node installation location")
+}
+
+func cmdRun(cmd *cobra.Command, args []string) (err error) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Using `context.Background()` is a temporary solution?

Reply:
No, I don't think so. Whats the problem with `context.Background()` here?

Reply:
You can use `cmd.Context()` here.

Reply:
Nothing :sweat_smile: I think that I was too asleep when I did the review and I thought that using background we didn't need to call `context.WithCancel`

Reply:
> You can use cmd.Context() here.

I don't see `Context()` in `cobra.Command` methods.

Reply:
Ah, nevermind it was `process.Ctx(cmd)`

Reply:
@mniewrzal Are you going to use `process.Ctx(cmd)` instead of `context.Background()` as @egonelbre  suggested?

Reply:
No, executing cmd with our `process.Exec()` is adding a lot of stuff that we don't need. For now I plan to stay as simple as possible. I can use only `process.Ctx(cmd)` to avoid few lines of code but I'm not sure if it's worth it.